lib/ansible_tower_client/api.rb

```diff
@@ -20,7 +20,10 @@ def version
     end
 
     def verify_credentials
-      JSON.parse(get("me/").body).fetch_path("results", 0, "username")
+      results = get("me/").body
+      JSON.parse(results).fetch_path("results", 0, "username")
+    rescue JSON::ParserError
+      raise AnsibleTowerClient::ConnectionError, "invalid host"
```

Comment:
Why `"invalid host"`?  That makes me think 4XX error.  Would "unexpected response" make more sense?

Reply:
thanks @bdunne